iroh/src/test_utils/qlog.rs

```diff
@@ -0,0 +1,142 @@
+//! Utils for emitting qlog files from iroh endpoint.
+
+use std::path::{Path, PathBuf};
+#[cfg(feature = "qlog")]
+use std::time::Instant;
+
+use n0_error::Result;
+use quinn::TransportConfig;
+#[cfg(feature = "qlog")]
+use quinn_proto::{QlogConfig, VantagePointType};
+
+/// Builder to create one or more related qlog configs.
+///
+/// This struct is available independently of feature flags, but if the "qlog" feature is not enabled
+/// it does not do anything.
+#[derive(Debug)]
+pub struct QlogFileGroup {
+    #[cfg(feature = "qlog")]
+    directory: PathBuf,
+    #[cfg(feature = "qlog")]
+    title: String,
+    #[cfg(feature = "qlog")]
+    start: Instant,
+}
+
+impl QlogFileGroup {
+    /// Creates a new [`QlogFileGroup] that is only enabled if feature flags and environment variables match.
+    ///
+    /// The qlog files will be written to `CARGO_MANIFEST_DIR/qlog`.
```

Comment:
Time to advertise https://crates.io/crates/testdir again. :)

Would be cool if we can use that by default.